### PR TITLE
Better logging for MSQ worker task

### PR DESCRIPTION
**Pull request by cryptoe**
Adding more logs statements to the worker implementation which makes it easier to debug.

<hr>

##### Key changed/added classes in this PR
 * `WorkerImpl`
 * `WorkerStageKernel`

<hr>

This PR has:

- [x] been self-reviewed.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:338]: Would recommend first emitting the info-level general comment. If debug is enabled, emit the work order as well. Else when searching for this message, one has to know what log level was set to know which message to search for.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:451]: Does the logger emit the task ID? If not, it might help to say what is stopping gracefully.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:528]: Nit: better to use standard English spacing: a space between "stage" and the bracket: `stage [%d]`.

The brackets are really only needed for items that can contain spaces, so we know what is the message and what is the value. But, this is a number. so `state %d`.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:591]: Finish received for what task?

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:598]: As above. Inconsistent use of colon. No colon is needed here: `stage %d`.

Does the logger automagically fill in the task ID? If not, would be good to include that so we can find events for a specific task.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:615]: Edit as above `stage %d, time chunk %d`

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/worker/WorkerStageKernel.java
Comment (paul-rogers) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/worker/WorkerStageKernel.java:215]: Same comments as above.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (cryptoe) [reply extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:598]: Each worker can only have one taskID and each log is generally task log is generally pulled from the druid console. 
In the case, people are using popular log agg tools, then they should add a dimension taskId to the "log line" via the constructs provided by the tool, "looking at file name in case of druid to figure out the taskId" is one such approach.
Hence I am not pushing taskId everywhere which just bloats the message.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java
Comment (cryptoe) [reply extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java:528]: In MSQ, in most places we are using this `[]` convention. I would hate to change it partially. Hence, if we decide to move away from using no brackets for numbers, then we should do it as part of another pr which touches all MSQ code.

**Comment by cryptoe** [comment]
@paul-rogers This PR is waiting on review/approval on you. Let me know if there are more things that are needed to be addressed.

**Review by paul-rogers — APPROVED** [review]
LGTM

**Pull request merged**
